ui/src/shared/components/CloudUpgradeButton.tsx

```diff
@@ -1,25 +1,56 @@
 // Libraries
 import React, {FC} from 'react'
 import {Link} from 'react-router'
+import {connect} from 'react-redux'
 
 // Components
 import CloudOnly from 'src/shared/components/cloud/CloudOnly'
 
 // Constants
 import {CLOUD_URL, CLOUD_CHECKOUT_PATH} from 'src/shared/constants'
+import {
+  HIDE_UPGRADE_CTA_KEY,
+  PAID_ORG_HIDE_UPGRADE_SETTING,
+} from 'src/cloud/constants'
 
-const CloudUpgradeButton: FC = () => {
+// Types
+import {AppState, OrgSetting} from 'src/types'
+
+interface StateProps {
+  show: boolean
```

Comment:
this is past style nit and bordering on a blocking suggestion. `show` reads like a verb: `show()`. it is not immediately clear what it represents and leaves some ambiguity about its intent.
```suggestion
  shouldShow: boolean
```

Reply:
`shouldShow` sounds too much like sh!tShow 🤣 

How about `inView`

Reply:
sounds good to me! you crack me up Timmy!